============================= test session starts ==============================
platform darwin -- Python 3.8.0, pytest-6.1.2, py-1.9.0, pluggy-0.13.1 -- /Users/meganfrenkel/Workspace/KERMit/server/venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/meganfrenkel/Workspace/KERMit
plugins: mock-3.3.1
collecting ... collected 109 items

server/tests/unit/test_authentication.py::test_authorize_token_missing_header PASSED [  0%]
server/tests/unit/test_authentication.py::test_authorize_token_missing_header2 PASSED [  1%]
server/tests/unit/test_authentication.py::test_authorize_token_malformed_header PASSED [  2%]
server/tests/unit/test_authentication.py::test_authorize_token_validation_error PASSED [  3%]
server/tests/unit/test_authentication.py::test_authorize_token_validation_success_register PASSED [  4%]
server/tests/unit/test_authentication.py::test_authorize_token_validation_success PASSED [  5%]
server/tests/unit/test_authentication.py::test_authorize_token_validation_not_registered PASSED [  6%]
server/tests/unit/test_authentication.py::test_register_already_registered PASSED [  7%]
server/tests/unit/test_authentication.py::test_register_missing_info_email PASSED [  8%]
server/tests/unit/test_authentication.py::test_register_missing_info_id PASSED [  9%]
server/tests/unit/test_authentication.py::test_register_error_googleauth PASSED [ 10%]
server/tests/unit/test_authentication.py::test_register_exception PASSED [ 11%]
server/tests/unit/test_authentication.py::test_register_user_not_registered PASSED [ 11%]
server/tests/unit/test_authentication.py::test_register_exception_db PASSED [ 12%]
server/tests/unit/test_google_auth.py::test_validate_token PASSED        [ 13%]
server/tests/unit/test_google_auth.py::test_get_user_information PASSED  [ 14%]
server/tests/unit/test_leaderboard.py::test_get_leaderboard_no_leaders PASSED [ 15%]
server/tests/unit/test_leaderboard.py::test_get_leaderboard_leaders_exist PASSED [ 16%]
server/tests/unit/test_player.py::test_save_commit PASSED                [ 17%]
server/tests/unit/test_player.py::test_save PASSED                       [ 18%]
server/tests/unit/test_player.py::test_find_all_puzzles_for_player PASSED [ 19%]
server/tests/unit/test_player.py::test_find_players_for_puzzle PASSED    [ 20%]
server/tests/unit/test_player.py::test_add_player_to_puzzle_already_too_many_players PASSED [ 21%]
server/tests/unit/test_player.py::test_add_player_to_puzzle_player_doesnt_exist PASSED [ 22%]
server/tests/unit/test_player.py::test_add_player_to_puzzle_ok PASSED    [ 22%]
server/tests/unit/test_player.py::test_to_player_to_str PASSED           [ 23%]
server/tests/unit/test_puzzle_exception.py::test_get_message PASSED      [ 24%]
server/tests/unit/test_puzzle_pieces.py::test_find_all_pieces PASSED     [ 25%]
server/tests/unit/test_puzzle_pieces.py::test_get_piece PASSED           [ 26%]
server/tests/unit/test_puzzle_pieces.py::test_get_piece_none PASSED      [ 27%]
server/tests/unit/test_puzzle_pieces.py::test_save_autocommit PASSED     [ 28%]
server/tests/unit/test_puzzle_pieces.py::test_save PASSED                [ 29%]
server/tests/unit/test_puzzle_pieces.py::test_update_success PASSED      [ 30%]
server/tests/unit/test_puzzle_pieces.py::test_update_fail_static PASSED  [ 31%]
server/tests/unit/test_puzzle_pieces.py::test_get_as_str PASSED          [ 32%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_valid_defaults PASSED [ 33%]
server/tests/unit/test_sudoku_puzzle.py::test_create_puzzle_valid_specification PASSED [ 33%]
server/tests/unit/test_sudoku_puzzle.py::test_create_puzzle_valid_difficulty_upper_bound PASSED [ 34%]
server/tests/unit/test_sudoku_puzzle.py::test_create_puzzle_valid_difficulty_lower_bound PASSED [ 35%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_invalid_difficulty_str PASSED [ 36%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_invalid_difficulty_too_low PASSED [ 37%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_invalid_difficulty_too_high PASSED [ 38%]
server/tests/unit/test_sudoku_puzzle.py::test_create_puzzle_valid_size_lower_bound PASSED [ 39%]
server/tests/unit/test_sudoku_puzzle.py::test_create_puzzle_valid_size_upper_bound PASSED [ 40%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_invalid_size_str PASSED [ 41%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_invalid_size_too_small PASSED [ 42%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_invalid_size_too_large PASSED [ 43%]
server/tests/unit/test_sudoku_puzzle.py::test_set_point_value1 PASSED    [ 44%]
server/tests/unit/test_sudoku_puzzle.py::test_set_point_value2 PASSED    [ 44%]
server/tests/unit/test_sudoku_puzzle.py::test_get_pieces_as_arr_all PASSED [ 45%]
server/tests/unit/test_sudoku_puzzle.py::test_get_pieces_as_arr_static_only PASSED [ 46%]
server/tests/unit/test_sudoku_puzzle.py::test_recreate_original_puzzle PASSED [ 47%]
server/tests/unit/test_sudoku_puzzle.py::test_check_for_completion_null_values PASSED [ 48%]
server/tests/unit/test_sudoku_puzzle.py::test_check_for_completion_complete PASSED [ 49%]
server/tests/unit/test_sudoku_puzzle.py::test_check_for_completion_incorrect_values PASSED [ 50%]
server/tests/unit/test_sudoku_puzzle.py::test_check_discrepancies_none PASSED [ 51%]
server/tests/unit/test_sudoku_puzzle.py::test_check_discrepancies_many_incorrect PASSED [ 52%]
server/tests/unit/test_sudoku_puzzle.py::test_check_discrepancies_incomplete_puzzle PASSED [ 53%]
server/tests/unit/test_sudoku_puzzle.py::test_get_puzzle_none PASSED     [ 54%]
server/tests/unit/test_sudoku_puzzle.py::test_get_puzzle_found PASSED    [ 55%]
server/tests/unit/test_sudoku_puzzle.py::test_save PASSED                [ 55%]
server/tests/unit/test_sudoku_puzzle.py::test_save_autocommit PASSED     [ 56%]
server/tests/unit/test_sudoku_puzzle.py::test_attempt_update_complete_puzzle PASSED [ 57%]
server/tests/unit/test_sudoku_puzzle.py::test_update_invalid_coordinate_y1 PASSED [ 58%]
server/tests/unit/test_sudoku_puzzle.py::test_update_invalid_coordinate_y2 PASSED [ 59%]
server/tests/unit/test_sudoku_puzzle.py::test_update_invalid_coordinate_x1 PASSED [ 60%]
server/tests/unit/test_sudoku_puzzle.py::test_update_invalid_coordinate_x2 PASSED [ 61%]
server/tests/unit/test_sudoku_puzzle.py::test_update_invalid_value_too_small PASSED [ 62%]
server/tests/unit/test_sudoku_puzzle.py::test_update_invalid_value_too_large PASSED [ 63%]
server/tests/unit/test_sudoku_puzzle.py::test_update_valid PASSED        [ 64%]
server/tests/unit/test_sudoku_puzzle.py::test_update_valid_lower_bounds PASSED [ 65%]
server/tests/unit/test_sudoku_puzzle.py::test_update_valid_upper_bounds PASSED [ 66%]
server/tests/unit/test_sudoku_puzzle.py::test_update_static_piece PASSED [ 66%]
server/tests/unit/test_sudoku_puzzle.py::test_update_valid_complete_puzzle PASSED [ 67%]
server/tests/unit/test_sudoku_puzzle.py::test_set_puzzle_complete PASSED [ 68%]
server/tests/unit/test_sudoku_puzzle.py::test_as_str PASSED              [ 69%]
server/tests/unit/test_sudoku_resources.py::test_sudoku_to_json PASSED   [ 70%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_none PASSED [ 71%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_one PASSED [ 72%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_many PASSED [ 73%]
server/tests/unit/test_sudoku_resources.py::test_sudoku_puzzles_create_known_exception PASSED [ 74%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_create_one_unknown_exception PASSED [ 75%]
server/tests/unit/test_sudoku_resources.py::test_create_puzzle_no_others PASSED [ 76%]
server/tests/unit/test_sudoku_resources.py::test_create_puzzle_unregistered_other PASSED [ 77%]
server/tests/unit/test_sudoku_resources.py::test_create_puzzle_registered_others PASSED [ 77%]
server/tests/unit/test_sudoku_resources.py::test_sudoku_puzzles_too_many_additional PASSED [ 78%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzle_none_retrieved PASSED [ 79%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzle_none_associated PASSED [ 80%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzle_found PASSED [ 81%]
server/tests/unit/test_sudoku_resources.py::test_join_sudoku_puzzle_already_joined PASSED [ 82%]
server/tests/unit/test_sudoku_resources.py::test_join_sudoku_puzzle PASSED [ 83%]
server/tests/unit/test_sudoku_resources.py::test_join_sudoku_puzzle_full PASSED [ 84%]
server/tests/unit/test_sudoku_resources.py::test_join_sudoku_puzzle_known_exception PASSED [ 85%]
server/tests/unit/test_sudoku_resources.py::test_join_sudoku_puzzle_unknown_exception PASSED [ 86%]
server/tests/unit/test_sudoku_resources.py::test_sudoku_puzzles_add_move_not_associated PASSED [ 87%]
server/tests/unit/test_sudoku_resources.py::test_sudoku_puzzles_add_move_success PASSED [ 88%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_add_position_invalid PASSED [ 88%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_add_value_invalid PASSED [ 89%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_add_move_exception PASSED [ 90%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_delete_piece PASSED [ 91%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_delete_piece_invalid PASSED [ 92%]
server/tests/unit/test_sudoku_resources.py::test_sudoku_puzzles_get_solution_not_associated PASSED [ 93%]
server/tests/unit/test_sudoku_resources.py::test_sudoku_puzzles_get_solution_not_associated_2 PASSED [ 94%]
server/tests/unit/test_sudoku_resources.py::test_sudoku_puzzles_get_solution PASSED [ 95%]
server/tests/unit/test_user.py::test_create_user PASSED                  [ 96%]
server/tests/unit/test_user.py::test_user_as_str PASSED                  [ 97%]
server/tests/unit/test_user.py::test_user_print PASSED                   [ 98%]
server/tests/unit/test_user.py::test_user_save PASSED                    [ 99%]
server/tests/unit/test_user.py::test_find_by_g_id PASSED                 [100%]

=============================== warnings summary ===============================
server/venv/lib/python3.8/site-packages/eventlet/patcher.py:1
  /Users/meganfrenkel/Workspace/KERMit/server/venv/lib/python3.8/site-packages/eventlet/patcher.py:1: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

server/venv/lib/python3.8/site-packages/dns/hash.py:23
  /Users/meganfrenkel/Workspace/KERMit/server/venv/lib/python3.8/site-packages/dns/hash.py:23: DeprecationWarning: dns.hash module will be removed in future versions. Please use hashlib instead.
    warnings.warn(

server/venv/lib/python3.8/site-packages/dns/namedict.py:35
  /Users/meganfrenkel/Workspace/KERMit/server/venv/lib/python3.8/site-packages/dns/namedict.py:35: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    class NameDict(collections.MutableMapping):

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================= 109 passed, 3 warnings in 2.35s ========================
